File: src/main/java/me/narutopig/bors/enchantments/enchantments/AftermathEnchant.java
package me.narutopig.bors.enchantments.enchantments;

import me.narutopig.bors.CustomEnchants;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

import static me.narutopig.bors.Util.hasEnchant;
import static me.narutopig.bors.Util.power;

public class AftermathEnchant implements Listener {
    public static int calculateDamage(Inventory inventory) {
        int res = 0;
        ItemStack[] contents = inventory.getContents();

        for (ItemStack itemStack : contents) {
            if (itemStack == null) continue;
            if (hasEnchant(itemStack, CustomEnchants.AFTERMATH)) {
                res += power(2, itemStack.getEnchantments().getOrDefault(CustomEnchants.AFTERMATH, 1) - 1);
            }
        }

        res = Math.min(res, 10);

        return res;
    }

    @EventHandler
    public void onDeath(PlayerDeathEvent event) {
        Player player = event.getEntity();

        if (player.getKiller() != null) {
            Player killer = player.getKiller();

            int amount = calculateDamage(player.getInventory());
            killer.damage(amount);
        }
    }
}

File: src/main/kotlin/io/github/narutopig/bors/enchantments/AftermathEnchant.kt
package io.github.narutopig.bors.enchantments

import io.github.narutopig.bors.util.Enchanting.getEnchantmentData
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent
import org.bukkit.inventory.Inventory

class AftermathEnchant : Listener {
    @EventHandler
    fun onDeath(event: PlayerDeathEvent) {
        val player = event.entity
        if (player.killer != null) {
            val killer = player.killer
            val amount = calculateDamage(player.inventory)
            killer!!.damage(amount.toDouble())
        }
    }

    companion object {
        fun calculateDamage(inventory: Inventory): Int {
            var res = 0
            val contents = inventory.contents
            for (itemStack in contents) {
                if (itemStack == null) continue
                val enchantmentData = getEnchantmentData(itemStack, CustomEnchants.AFTERMATH.name)
                res += enchantmentData.level
            }
            res = res.coerceAtMost(10)
            return res
        }
    }
}